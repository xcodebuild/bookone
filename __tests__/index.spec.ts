import shell from 'shelljs';
import path from 'path';
import fs from 'fs-extra-promise';

const EXAMPLE_DIR = path.join(__dirname, 'fixture/example');
const EXAMPLE_DIST_DIR = path.join(EXAMPLE_DIR, 'dist');
const BOOKONE_BIN = path.join(__dirname, '../bin/bookone');

const cleanup = () => {
    if (fs.existsSync(EXAMPLE_DIST_DIR)) {
        shell.rm(EXAMPLE_DIST_DIR);
    }
}

function readFileContent(filePath: string) {
    return fs.readFileSync(filePath, 'utf-8');
}

describe('CLI', () => {
    beforeEach(cleanup);
    
    it('build works', () => {
        const command = `cd ${EXAMPLE_DIR} && ${BOOKONE_BIN} build`;
        shell.exec(command);
        const lsResult = shell.ls(EXAMPLE_DIST_DIR);
        expect(lsResult).toContain('chapter-one');
        expect(lsResult).toContain('chapter-two');
    });

    it('get title from markdown', () => {
        const command = `cd ${EXAMPLE_DIR} && ${BOOKONE_BIN} build`;
        shell.exec(command);
        // get Title
        expect(readFileContent(path.join(EXAMPLE_DIST_DIR, 'chapter-one/intro.html'))).toContain('Before Title');
    });

    it('build with -t', () => {
        const command = `cd ${EXAMPLE_DIR} && ${BOOKONE_BIN} build -t BookTitle`;
        shell.exec(command);
        // get Title
        expect(readFileContent(path.join(EXAMPLE_DIST_DIR, 'chapter-one/intro.html'))).toContain('<title>BookTitle</title>');
    });
});