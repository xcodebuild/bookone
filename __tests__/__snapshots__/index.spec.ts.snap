// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI image cite 1`] = `"https://i.postimg.cc/0zQTZkZx/5436704-50.png#xcodebuild"`;

exports[`CLI image cite 2`] = `"Figure 2-2"`;

exports[`CLI image title will be display 1`] = `"/public/images/mock.png#mock"`;

exports[`CLI image title will be display 2`] = `
"Figure 2-1: 
            Image Title
        
            Figure 2-2: 
            Image Cite"
`;

exports[`CLI image url repect baseurl 1`] = `"/baseurl/public/images/mock.png#mock"`;

exports[`CLI image url repect baseurl 2`] = `
"Figure 2-1: 
            Image Title"
`;

exports[`CLI internal link 1`] = `undefined`;

exports[`CLI public files copy to dist 1`] = `
"console.log('mock js');
"
`;
